converters/fromZigbee.js

```diff
@@ -2595,6 +2595,26 @@ const converters = {
             return result;
         },
     },
+    ts0219_siren: {
+        cluster: 'ssIasZone',
+        type: 'attributeReport',
+        convert: (model, msg, publish, options, meta) => {
+            const zoneStatus = msg.data.zoneStatus;
+            return {
+                alarm: zoneStatus === 17 ? true : false,
+            };
+        },
+    },
+    ts0219_power_source: {
+        cluster: 'genBasic',
+        type: 'attributeReport',
+        convert: (model, msg, publish, options, meta) => {
+            const powerSource = msg.data.powerSource;
+            return {
+                ac_connected: powerSource === 2 ? true : false,
```

Comment:
Do I understand correctly that this device can change it's mode from ac powered to battery powered?

Reply:
yes this device can work on battery, for almost one day (not more but in case of power shutdown it's enough...)
And for me it's important to know if the device work on battery or is plugged into the wall :) 
I take this part of code from another issue (and of course i check it work !) : https://github.com/Koenkk/zigbee-herdsman-converters/issues/1822